File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    for i in range(1, yellow + 1):
        x = i
        y = int(yellow // i)
        if x * y == yellow:
            if (x + y) == int((brown - 4) // 2):
                return [y+2, x+2]
    

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        int total = yellow + brown;
        
        while(true){
            for(int i = 3; i < total; i++){
                int y = i;
                int x = total / y;
                if (x >= y && (x-2) * (y-2) == yellow){
                    return new int[]{x,y};
                }
            }
            break;
        }
        int[] answer = {};
        
        return answer;
    }
}